nucypher/blockchain/eth/trackers/dkg.py

```diff
@@ -123,7 +131,17 @@ def __init__(
         )
 
         self.task = EventScannerTask(scanner=self.scan)
-        self._participation_states = dict()  # { ritual_id -> ParticipationState }
+
+        cache_ttl = (
+            self.coordinator_agent.get_timeout()
+            + self._RITUAL_TIMEOUT_ADDITIONAL_TTL_BUFFER
+        )
+        self._participation_states = TTLCache(
```

Comment:
Ah yes, return of the `TTLCache` :-)

Reply:
Haha yes - you remembered! I indeed revived it from the old Porter WIP PR, and made some improvements.